sdk/core/amqp-common/src/requestResponseLink.ts

```diff
@@ -122,25 +119,56 @@ export class RequestResponseLink implements ReqResLink {
           request.message_id = generate_uuid();
         }
 
+        const rejectOnAbort = () => {
+          const address = this.receiver.address || "address";
+          const desc: string =
+            `[${this.connection.id}] The request with message_id ` +
+            `"${request.message_id}" to "${address}" has been cancelled by the user.`;
```

Comment:
The message_id is something that we make up in this library, the end user will have no knowledge of it. So this message will not help the user.

Also, as per https://github.com/Azure/azure-sdk-for-js/issues/3601 we are leaning towards a standard error `AbortError` for cancellation.

Therefore, I suggest the below:
- On cancellation throw an error with name `AbortError`
- Let the caller update the error message as it has more knowledge about the call that is being cancelled. Or update `SendRequestOptions` to take in another option called `requestName` and use that in the error message